src/OrchardCore.Modules/OrchardCore.Taxonomies/Drivers/TaxonomyContentsAdminListDisplayDriver.cs

```diff
@@ -46,8 +51,18 @@ public override async Task<IDisplayResult> EditAsync(ContentOptionsViewModel mod
                 return null;
             }
 
+            var taxonomyContentItemIds = settings.TaxonomyContentItemIds;
+            if (!string.IsNullOrWhiteSpace(model.SelectedContentType))
+            {
+                var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition(model.SelectedContentType);
+                var fieldDefinitions = contentTypeDefinition
+                    .Parts.SelectMany(x => x.PartDefinition.Fields.Where(f => f.FieldDefinition.Name == nameof(TaxonomyField)));
+                var fieldTaxonomyContentItemIds = fieldDefinitions.Select(x => x.GetSettings<TaxonomyFieldSettings>().TaxonomyContentItemId);
+                taxonomyContentItemIds = taxonomyContentItemIds.Intersect(fieldTaxonomyContentItemIds).ToArray();
+            }
+
```

Comment:
This should just do a check here for `if (!taxonomyContentItemIds.Any())` `return null;`